src/data/localStorage.js

```diff
@@ -0,0 +1,129 @@
+ * @return {object} array of post objects
+ */
+export const getLocalStorageFeed = () => {
+  const localFeed = JSON.parse(localStorage.getItem(localStorageFeedKey));
+  let pinStatus = false;
+  if (localFeed) {
+    let dataFeed = [];
+    localFeed.forEach((item) => {
+      const localPost = JSON.parse(localStorage.getItem(item));
+      if (localPost.isPinned) pinStatus = true;
+      dataFeed.push(localPost);
+    });
+    return { feed: dataFeed, pinStatus: pinStatus };
+  }
+  return null;
+};
+
+/**
+ * setLocalStorageFeed() persists the given array to localStorage at localStorageFeedKey
+ * @param {array} localFeed array of post ref strings
+ */
+export const setLocalStorageFeed = (localFeed) => {
+  localStorage.setItem(localStorageFeedKey, JSON.stringify(localFeed));
+};
+
+/**
+ * insertLocalStorageFeed() updates the local storage feed item when post created
+ * @param {string} ref post ref to be added to local storage
+ */
+export const insertLocalStorageFeed = (ref) => {
+  let currentLocalFeed = JSON.parse(localStorage.getItem(localStorageFeedKey));
+  if (currentLocalFeed) {
+    currentLocalFeed.unshift(ref);
+  } else {
+    currentLocalFeed = [ref];
+  }
+  localStorage.setItem(localStorageFeedKey, JSON.stringify(currentLocalFeed));
+};
+
+/**
+ * clearLocalStorageFeed() updates the local storage feed with item when post created
+ */
+export const clearLocalStorageFeed = () => {
+  localStorage.setItem(localStorageFeedKey, null);
+};
+
+/**
+ * getLocalStoragePost() get a post from local storage
+ * @param {string} ref post ref to retrieve
+ * @return {object} post from local storage
+ */
+export const getLocalStoragePost = (ref) => {
+  const localPost = JSON.parse(localStorage.getItem(ref));
+  return localPost;
+};
+
+/**
+ * setLocalStoragePost() persist a post's data to local storage
+ * @param {string} ref post ref
+ * @param {object} postJSON post data
+ */
+export const setLocalStoragePost = (ref, postJSON) => {
+  let storageJSON = {
+    ts: postJSON.ts,
+    isPinned: postJSON.isPinned,
+    ref: ref,
+    content: {
+      title: postJSON.content.title,
+      text: postJSON.content.text,
+      previewImage: getPreviewImage(postJSON.content.text),
+      ext: {
+        subtitle: postJSON.content.ext.subtitle,
+        resolverSkylink: postJSON.content.ext.resolverSkylink,
+        postPath: postJSON.content.ext.postPath,
+      },
+    },
+  };
+  localStorage.setItem(ref, JSON.stringify(storageJSON));
+};
+
+/**
+ * editLocalStoragePost() add a post's data to local storage
+ * @param {string} title updated title
+ * @param {string} subtitle updated subtitle
+ * @param {string} blogMD updated blog body
+ * @param {boolean} isPinned updated pinned val
+ * @param {string} ts updated timestamp
+ * @param {string} ref post ref
+ */
+export const editLocalStoragePost = (
+  title,
+  subtitle,
+  blogMD,
+  isPinned,
+  ts,
+  ref
+) => {
+  let postJSON = JSON.parse(localStorage.getItem(ref));
```

Comment:
If there is no entry, will this return an empty object? 

We should probably also add null checks to the input params to avoid overwritting data. Either we should enforce that all the fields are set, or conditionally set the fields if they are not null.

If we need to delete a local storage item we should have a specific method for that. So edit should protect against accidental deletion.

Reply:
also this can be DRY'd up by calling `getLocalStoragePost`